### PR TITLE
Autocomplete support

### PR DESCRIPTION
**Pull request by pbitty**
This adds autocomplete support, available in the latest version of [go-langserver](https://github.com/sourcegraph/go-langserver).

### CODE REVIEW

File: lib/ide-go.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const { spawn } = require('child_process')
 const { AutoLanguageClient } = require('atom-languageclient')
+const AutocompleteAdapter = require('../node_modules/atom-languageclient/build/lib/adapters/autocomplete-adapter')
```
Comment (ckaznocha) [inline lib/ide-go.js]: I'm not sure this is needed?

File: package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "atom": ">=1.0.0 <2.0.0"
   },
   "dependencies": {
-    "atom-languageclient": "^0.6.1",
+    "atom-languageclient": "^0.7.0",
```
Comment (ckaznocha) [inline package.json]: I had another PR that updated the deps that I finally merged. Mind just pulling master into your branch?

**Review by ckaznocha — CHANGES_REQUESTED** [review]
Thanks for the PR @pbitty this is great! I just had some pretty minor nit picks. I'm psyched to get this merged!

**Comment by pbitty** [comment]
@ckaznocha, changes addressed.

I have a [PR open](https://github.com/sourcegraph/go-langserver/pull/221) on `go-langserver` to add func snippets.  There are still some pieces to be decided.  If it is merged, there may be some tweaks needed to `ide-go` to support it:
* If the feature is disabled by default, a CLI flag will needed enable it
* If `InsertText` is not provided (because it is deprecated), it breaks snippet support.  I have a [PR open on atom-languageclient](https://github.com/atom/atom-languageclient/pull/165) to fix this

It could also end up being merged in a state that requires on changes here.  Either way, I am happy to submit a new PR depending on the outcome.

**Review by ckaznocha — APPROVED** [review]

**Comment by pbitty** [comment]
@ckaznocha Will you be merging this soon?

sourcegraph/go-langserver#221 was merged in a way that does not require any changes here.  It will support autocomplete and along with func arg snippets.

**Pull request merged**